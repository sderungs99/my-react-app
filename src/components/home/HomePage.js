import React, { Component } from 'react';
import './HomePage.css';
import Button from '../common/Button';

class HomePage extends Component {

    constructor(props) {
        super(props);

        this.state = {
            clickCounter: 0
        };

        this.handleClick = this.handleClick.bind(this);
    }

    handleClick() {
        this.setState({
            clickCounter: this.state.clickCounter + 1
        });
    }

    render() {
        return (
            <div>
                <p className="App-intro">
                    To get started, edit <code>src/App.js</code> and save to reload.
                </p>
                <p><Button onClick={this.handleClick} buttonLabel="Test Button" /></p>
                <p>Click Counter: {this.state.clickCounter}</p>
            </div>
        );
    }
}

export default HomePage;