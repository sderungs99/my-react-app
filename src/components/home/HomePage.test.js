import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import HomePage from './HomePage';

function setup(props) {
    return shallow(<HomePage {...props} />);
}

it('renders without crashing', () => {
    setup();
});

it('renders as expected', () => {
    const component = renderer.create(
        <HomePage />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('adds up button clicks correctly', () => {
    const wrapper = setup();
    const expectedClickCounter = wrapper.instance().state.clickCounter + 1;
    wrapper.instance().handleClick();
    expect(wrapper.instance().state.clickCounter).toBe(expectedClickCounter);
});