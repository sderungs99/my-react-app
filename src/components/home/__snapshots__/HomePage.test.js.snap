// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
  <p>
    <button
      onClick={[Function]}
    >
      Test Button
    </button>
  </p>
  <p>
    Click Counter: 
    0
  </p>
</div>
`;
